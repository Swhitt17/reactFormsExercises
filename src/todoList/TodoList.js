import React,{useState} from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

import {v4 as uuid} from "uuid"

const TodoList = () => {

const [todos,setTodos] = useState([]);
const addTodo = (newTodo) => {
    setTodos(todos => [...todos, {...newTodo, id:uuid()}])
}

const removeTodo = id => {
    setTodos(todos => todos.filter(todo => todo.id !== id));
}

return (
    <div>
        <h2>Todo List</h2>
        <NewTodoForm addTodo={addTodo}/>
        <div>
            {todos.map( todo => ( 
            <Todo key={todo.id} id={todo.id} task={todo.task} handleRemove={removeTodo}/>))}
        </div> 
    </div>
)
}

export default TodoList;