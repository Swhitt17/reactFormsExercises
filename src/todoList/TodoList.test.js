import React from "react";
import {render, fireEvent} from "@testing-library/react"
import TodoList from "./TodoList";

function addTodo(todoList,task = "Wash Dishes"){
    const input = todoList.getByLabelText("Task");
    fireEvent.change(input, {target: {value: task}});
     const btn = todoList.getByText("Submit");
     fireEvent.click(btn);
}

test("it renders without crashing", () => {
    render(<TodoList />);
});

test("it matches snapshot", () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});


test("it add new todo", () => {
    const list = render(<TodoList />);
    addTodo(list);
    expect(list.getByText('Wash Dishes')).toBeInTheDocument();
   
});
