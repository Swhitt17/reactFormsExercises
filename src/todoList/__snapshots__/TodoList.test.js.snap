// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Todo List
    </h2>
    <form>
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
